Empty annotations cause IngressRouter to hit nil crash

### Describe the bug

Having an empty set of annotations causes a crash. Stack trace:
```
panic: assignment to entry in nil map

goroutine 213 [running]:
github.com/fluxcd/flagger/pkg/router.filterMetadata(...)
	/workspace/pkg/router/util.go:32
github.com/fluxcd/flagger/pkg/router.(*IngressRouter).makeAnnotations(0xc00086c180, 0xc000be5860)
	/workspace/pkg/router/ingress.go:217 +0x5a
github.com/fluxcd/flagger/pkg/router.(*IngressRouter).Reconcile(0xc00086c180, 0xc0006d8000)
	/workspace/pkg/router/ingress.go:81 +0x965
github.com/fluxcd/flagger/pkg/controller.(*Controller).advanceCanary(0xc0004f4dc0, {0xc000a61056, 0xa}, {0xc000a61060, 0xe})
	/workspace/pkg/controller/scheduler.go:225 +0x746
github.com/fluxcd/flagger/pkg/controller.CanaryJob.Start.func1()
	/workspace/pkg/controller/job.go:35 +0x3a
created by github.com/fluxcd/flagger/pkg/controller.CanaryJob.Start
	/workspace/pkg/controller/job.go:33 +0xae
```

### To Reproduce

1. Server-side apply a manifest with the empty set of annotations
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations: {}
```
2. Deploy out a canary CRD

### Expected behavior

Canary should proceed, or log its failure. It's worth noting, once I add any annotation, Initialization completed.

### Additional context

- Flagger version: `ghcr.io/fluxcd/flagger:1.21.0`
- Kubernetes version:   1.21
- Service Mesh provider:
- Ingress provider: ingress-nginx

--- a/pkg/router/util.go
+++ b/pkg/router/util.go
@@ -29,6 +29,9 @@ func includeLabelsByPrefix(labels map[string]string, includeLabelPrefixes []stri
 }
 
 func filterMetadata(meta map[string]string) map[string]string {
+	if meta == nil {
+		meta = map[string]string{}
+	}
 	// prevent Flux from overriding Flagger managed objects
 	meta[toolkitReconcileKey] = toolkitReconcileValue
 	meta[helmDriftDetectionKey] = toolkitReconcileValue

